### ISSUE TITLE
Heisenbug in fake_labels

### ISSUE THREAD

**Issue by thcrock** [dssg/triage#138]
When this code (in tests/utils.py):

```
def fake_labels(length):
    return numpy.array([random.choice([True, False]) for i in range(0, length)])
```

Comes up with all the same values, some of the metric calculators error out. We should just hardcode this to have some variation.

**Comment by thcrock** [comment]
Moved to https://github.com/dssg/architect/issues/13

**Comment by thcrock** [comment]
Not fixed (or rather, should have been reopened when architect merged back in here)

**Comment by ecsalomon** [comment]
Per yesterday's conversation:

- The best first version of this is probably to catch the errors, print an informative warning, and insert `NULL` `value`s for the affected metrics.

- Having the `NULL`s will cause audition to blow up if the model groups are included in its query, but, generally, users should not be trying to run audition after experiencing this issue, so handling this in audition is a nice-to-have but not necessary change.

**Comment by ecsalomon** [comment]
Hmmm, as I am addressing the subsets issue, I thought this might also be the correct way to handle empty subsets, so the "evaluations" still get written to the database but with the relevant information that there were no labels to evaluate.

**Comment by thcrock** [comment]
Closed in #552
